src/user/rust.rs

```diff
@@ -0,0 +1,198 @@
+use parse_display::{Display, FromStr};
+use rust_nix_templater::{options::RustToolchainChannel, *};
+
+use crate::ActionStack;
+
+use super::{UserAction, UserMetadata, UserPrompt};
+
+#[derive(Debug, Clone, PartialEq, Eq, Display, FromStr)]
+pub(crate) enum Prompt {
+    #[display("generate flake")]
+    Generate,
+    #[display("set package name ({0})")]
+    SetPackageName(String),
+    #[display("set executable name ({0})")]
+    SetExecName(String),
+    #[display("set description ({0})")]
+    SetDescription(String),
+    #[display("set toolchain ({0})")]
+    SetToolchain(RustToolchainChannel),
+    #[display("set license ({0})")]
+    SetLicense(String),
```

Comment:
we prefill this usually, no need to always allocate.
```suggestion
    SetLicense(std::borrow::Cow<'static, str>),
```